File: MicroTube.Client.Angular/src/app/services/ICommentingService.ts
import { Observable } from "rxjs";
import { CommentDto } from "../data/Dto/CommentDto";
import { CommentRequestDto } from "../data/Dto/CommentRequestDto";
import { EditCommentRequestDto } from "../data/Dto/EditCommentRequestDto";

export interface ICommentingService
{
  comment(videoId: string, request: CommentRequestDto): Observable<CommentDto>;
  editComment(commentId: string, request: EditCommentRequestDto): Observable<CommentDto>;
  deleteComment(commentId: string): Observable<CommentDto>;
}

File: MicroTube/Services/Comments/ICommentingService.cs
using MicroTube.Data.Models.Comments;
namespace MicroTube.Services.Comments
{
	public interface ICommentingService
	{
		Task<IServiceResult<IComment>> Comment(string userId, string targetId, string content);
		Task<IServiceResult<IComment>> DeleteComment(string userId, string commentId);
		Task<IServiceResult<IComment>> EditComment(string userId, string newContent, string commentId);
	}
}